src/lib/core/hooks/computeDefaultDependentAxisRange.ts

```diff
@@ -0,0 +1,251 @@
+    if (plotType == null || plotType == 'Histogram')
+      return histogramDefaultDependentAxisMinMax(
+        data as PromiseHookState<
+          HistogramDataWithCoverageStatistics | undefined
+        >
+      );
+    else if (plotType == 'Barplot')
+      // barplot only computes max value
+      return {
+        min: 0,
+        max: barplotDefaultDependentAxisMax(
+          data as PromiseHookState<BarplotDataWithStatistics | undefined>
+        ),
+      };
+    // boxplot
+    else if (plotType === 'Boxplot')
+      return boxplotDefaultDependentAxisMinMax(
+        data as PromiseHookState<BoxplotDataWithCoverage | undefined>,
+        yAxisVariable
+      );
+  }, [data]);
+
+  // set useMemo to avoid infinite loop
+  // set default dependent axis range for better displaying tick labels in log-scale
+  const defaultDependentAxisRange = useMemo(() => {
+    if (plotType === 'Histogram' || plotType === 'Barplot')
+      return defaultDependentAxisMinMax?.min != null &&
+        defaultDependentAxisMinMax?.max != null
+        ? {
+            // set min as 0 (count, proportion) for non-logscale for histogram/barplot
+            min:
+              (vizConfig as HistogramConfig | BarplotConfig).valueSpec ===
+              'count'
+                ? 0
+                : (vizConfig as HistogramConfig | BarplotConfig)
+                    .dependentAxisLogScale
+                ? // determine min based on data for log-scale at proportion
+                  // need to check defaultDependentAxisMinMax.min !== 0
+                  defaultDependentAxisMinMax.min !== 0 &&
+                  defaultDependentAxisMinMax.min < 0.001
+                  ? defaultDependentAxisMinMax.min * 0.8
+                  : 0.001
+                : 0,
+            max: defaultDependentAxisMinMax.max * 1.05,
+          }
+        : undefined;
+    // boxplot
+    else if (plotType === 'Boxplot')
+      return defaultDependentAxisMinMax?.min != null &&
+        defaultDependentAxisMinMax?.max != null
+        ? {
+            min: defaultDependentAxisMinMax.min,
+            max: defaultDependentAxisMinMax.max * 1.05,
+          }
+        : undefined;
+  }, [
+    defaultDependentAxisMinMax,
+    (vizConfig as HistogramConfig | BarplotConfig).valueSpec,
+    (vizConfig as HistogramConfig | BarplotConfig).dependentAxisLogScale,
+  ]);
+
+  // use this to avoid infinite loop: also use useLayoutEffect to avoid async/ghost issue
+  const updateVizConfigRef = useRef(updateVizConfig);
+  useLayoutEffect(() => {
+    updateVizConfigRef.current = updateVizConfig;
+  }, [updateVizConfig]);
+
+  // update vizConfig.dependentAxisRange as it is necessary for set range correctly
+  useLayoutEffect(() => {
+    if (!data.pending)
+      updateVizConfigRef.current({
+        dependentAxisRange: defaultDependentAxisRange,
+      });
+    else
+      updateVizConfigRef.current({
+        dependentAxisRange: undefined,
+      });
+  }, [data, defaultDependentAxisRange]);
```

Comment:
As far as I can tell, this section is not needed (and no need to pass `updateVizConfig` as a prop).
I have tested with bar, scatter and boxplot, and the dependent axis seems to behave properly, and the vizConfigs are saved properly too.

Please could you describe what doesn't work (if any) if this code is removed?